pkgs/by-name/fl/flclash/package.nix

```diff
@@ -12,21 +12,20 @@
 
 let
   pname = "flclash";
-  version = "0.8.90";
-
-  src =
-    (fetchFromGitHub {
-      owner = "chen08209";
-      repo = "FlClash";
-      tag = "v${version}";
-      hash = "sha256-wEgWjzdP7HeWgDacaP9fYNczG9BrTN790AQ5aj9scwM=";
-      fetchSubmodules = true;
-    }).overrideAttrs
-      (_: {
-        GIT_CONFIG_COUNT = 1;
-        GIT_CONFIG_KEY_0 = "url.https://github.com/.insteadOf";
-        GIT_CONFIG_VALUE_0 = "git@github.com:";
-      });
+  version = "0.8.91";
+
+  src = fetchFromGitHub {
+    owner = "chen08209";
+    repo = "FlClash";
+    tag = "v${version}";
+    preFetch = ''
```

Comment:
So far [most packages in Nixpkgs](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs%20GIT_CONFIG_COUNT&type=code) uses `overrideAttrs`.  Is there a reason to change it?

Reply:
Not using overrideAttrs is actually an improvment. Also not always is tha majority code correct.

Reply:
> It seems to no longer be effective

`fetchurl` recently enabled `__structuredAttrs`. This means derivation attrs are made available as shell variables, but not exported as environment variables.

The correct fix is to define them in `env`.